google-cloud-firestore/src/test/java/com/google/cloud/firestore/ToStringTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.firestore.spi.v1.FirestoreRpc;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/** @author Eran Leshem */
+@RunWith(MockitoJUnitRunner.class)
+public class ToStringTest {
+
+  @Spy
+  private final FirestoreImpl firestoreMock =
+      new FirestoreImpl(
+          FirestoreOptions.newBuilder().setProjectId("test-project").build(),
+          Mockito.mock(FirestoreRpc.class));
+
+  private WriteBatch batch;
+  private DocumentReference documentReference;
+
+  @Before
+  public void before() {
+    batch = firestoreMock.batch();
+    documentReference = firestoreMock.document("coll/doc");
+  }
+
+  @Test
+  public void testDocumentSnapshot() {
+    Assert.assertEquals(
+        "DocumentSnapshot{docRef=DocumentReference{path=projects/test-project/databases/(default)"
+            + "/documents/coll/doc}, fields={key=string_value: \"value\"\n"
+            + "}, readTime=null, updateTime=null, createTime=null}",
```

Comment:
This test has a whole bunch of "nulls" in the toString() output. Could you instead use a DocumentSnapshot that uses non-null values? This will help ensure that the right text is associated with each key.